src/looprestoration.rs

```diff
@@ -294,3 +298,106 @@ pub(crate) unsafe fn padding<BD: BitDepth>(
         }
     };
 }
+
+// TODO(randompoison): Temporarily public until init logic is deduplicated.
+pub(crate) unsafe extern "C" fn wiener_erased<BD: BitDepth>(
```

Comment:
Should this be named `wiener_c`?  Since part of its purpose is to be `extern "C"`, which `wiener_rust` doesn't need to.  But it's also doing the type erasure, so maybe it should be `wiener_c_erased`?

Reply:
I wanted to use the `_erased` naming convention that I started using in #332 for the type-erased wrapper functions. I thought more generally we were replacing the `_c` suffix with `_rust`?

Reply:
I was keeping the `*_c` for `fn`s that have to be `extern "C"`.  The `*_erased` convention is useful, too, though, so I think `wiener_c_erased` might be good here.

Reply:
I think that's a bit redundant, since `_erased` is only going to be used for functions that are `extern "C"` callbacks. Do we need that `_c` suffix?

Reply:
I think it's just useful for searching.